test/linalg/tridiag.jl

```diff
@@ -228,118 +16,180 @@ end
+        if elty == Int
+            srand(61516384)
+            d = rand(1:100, n)
+            dl = -rand(0:10, n-1)
+            du = -rand(0:10, n-1)
+            v = rand(1:100, n)
+            B = rand(1:100, n, 2)
+            a = rand(1:100, n-1)
+            b = rand(1:100, n)
+            c = rand(1:100, n-1)
+        else
+            d = convert(Vector{elty}, 1. + randn(n))
+            dl = convert(Vector{elty}, randn(n - 1))
+            du = convert(Vector{elty}, randn(n - 1))
+            v = convert(Vector{elty}, randn(n))
+            B = convert(Matrix{elty}, randn(n, 2))
+            a = convert(Vector{elty}, randn(n - 1))
+            b = convert(Vector{elty}, randn(n))
+            c = convert(Vector{elty}, randn(n - 1))
             if elty <: Complex
-                a += im*convert(Vector{elty}, randn(n))
-                b += im*convert(Vector{elty}, randn(n-1))
+                a += im*convert(Vector{elty}, randn(n - 1))
+                b += im*convert(Vector{elty}, randn(n))
+                c += im*convert(Vector{elty}, randn(n - 1))
             end
+        end
+        ε = eps(abs2(float(one(elty))))
+        @test_throws DimensionMismatch SymTridiagonal(dl, ones(elty, n + 1))
+        @test_throws ArgumentError SymTridiagonal(rand(n, n))
+        @test_throws ArgumentError Tridiagonal(dl, dl, dl)
+        @test_throws ArgumentError convert(SymTridiagonal{elty}, Tridiagonal(dl, d, du))
+
+        if elty != Int
+            @testset "issue #1490" begin
+                @test det(ones(elty,3,3)) ≈ zero(elty) atol=3*eps(real(one(elty)))
+                @test det(SymTridiagonal(elty[],elty[])) == one(elty)
+            end
+        end
 
-            @test_throws DimensionMismatch SymTridiagonal(a, ones(elty, n+1))
-            @test_throws ArgumentError SymTridiagonal(rand(n,n))
+        @testset "constructor" begin
+            for (x, y) in ((d, dl), (GenericArray(d), GenericArray(dl)))
+                ST = (SymTridiagonal(x, y))::SymTridiagonal{elty, typeof(x)}
+                @test ST == Matrix(ST)
+                @test ST.dv === x
+                @test ST.ev === y
+                TT = (Tridiagonal(y, x, y))::Tridiagonal{elty, typeof(x)}
+                @test TT == Matrix(TT)
+                @test TT.dl === y
+                @test TT.d  === x
+                @test TT.du === y
+            end
+            # enable when deprecations for 0.7 are dropped
+            # @test_throws MethodError SymTridiagonal(dv, GenericArray(ev))
+            # @test_throws MethodError SymTridiagonal(GenericArray(dv), ev)
+            # @test_throws MethodError Tridiagonal(GenericArray(ev), dv, GenericArray(ev))
+            # @test_throws MethodError Tridiagonal(ev, GenericArray(dv), ev)
+        end
+        @testset "interconversion of Tridiagonal and SymTridiagonal" begin
+            @test Tridiagonal(dl, d, dl) == SymTridiagonal(d, dl)
+            @test SymTridiagonal(d, dl) == Tridiagonal(dl, d, dl)
```

Comment:
This test is completely equivalent with the one on the line above.

Reply:
A bit of a theme in this file...